src/Centreon/Application/Controller/AcknowledgementController.php

```diff
@@ -596,21 +598,109 @@ public function massDisacknowledgeResources(
+     * @return View
+     * @throws \Exception
      */
-    private function validateResource(
-        EntityValidator $validator,
-        ResourceEntity $resource,
-        array $contextGroups
-    ): ConstraintViolationListInterface {
-        return $validator->validate(
-            $resource,
-            null,
-            $contextGroups
+    public function massAcknowledgeResources(
+        Request $request,
+        EntityValidator $entityValidator,
+        SerializerInterface $serializer
+    ): View {
+        $this->denyAccessUnlessGrantedForApiRealtime();
+
+        /**
+         * @var Contact $contact
+         */
+        $contact = $this->getUser();
+
+        /**
+         * @var AckRequest $ackRequest
+         */
+        $ackRequest = $serializer->deserialize(
+            (string)$request->getContent(),
+            AckRequest::class,
+            'json'
         );
+
+        $this->acknowledgementService->filterByContact($contact);
+
+        //validate input
+        $errorList = new ConstraintViolationList();
+
+        //validate resources
+        $resources = $ackRequest->getResources() ?? [];
+        foreach ($resources as $resource) {
+            if ($resource->getType() === ResourceEntity::TYPE_SERVICE) {
+                $errorList->addAll(ResourceService::validateResource(
+                    $entityValidator,
+                    $resource,
+                    ResourceEntity::VALIDATION_GROUP_DISACK_SERVICE
+                ));
+            } elseif ($resource->getType() === ResourceEntity::TYPE_HOST) {
+                $errorList->addAll(ResourceService::validateResource(
+                    $entityValidator,
+                    $resource,
+                    ResourceEntity::VALIDATION_GROUP_DISACK_HOST
+                ));
+            } else {
+                throw new \RestBadRequestException('Incorrect resource type for acknowledgement');
+            }
+        }
+
+        //validate acknowledgement
+        $acknowledgement = $ackRequest->getAcknowledgement();
+        $errorList->addAll(
+            $entityValidator->validate(
+                $acknowledgement,
+                null,
+                Acknowledgement::VALIDATION_GROUP_ACK_RESOURCE
+            )
+        );
+
+        if ($errorList->count() > 0) {
+            throw new ValidationFailedException($errorList);
+        }
+
+        //set default values [sticky, persistent_comment] to true
+        $acknowledgement->setSticky(true);
+        $acknowledgement->setPersistentComment(true);
+
+        foreach ($resources as $resource) {
+            //start acknowledgement process
+            try {
+                if ($this->hasAckRightsForResource($contact, $resource)) {
+                    $this->acknowledgementService->acknowledgeResource(
+                        $resource,
+                        $acknowledgement
+                    );
+                }
+            } catch (\Exception $e) {
+                continue;
```

Comment:
Perhaps we can display an error to inform the user that certain resources were not acknowledged ?

Reply:
That has already been discussed and there is a ticket for it to be implemented after the release, because we need to handle this on the frontend as well, and discuss with them about structure of warning etc.. But thanks for the idea, we will definitely do this in a way it can be reused in the future for similar cases.

Reply:
Perfect.